pkg/kubectl/resource/visitor.go

```diff
@@ -44,7 +42,7 @@ type Visitor interface {
 }
 
 // VisitorFunc implements the Visitor interface for a matching function
-type VisitorFunc func(*Info) error
+type VisitorFunc func(*Info, error) error
```

Comment:
I haven't had time to do a review, but I took a quick glance.  Can you add some godoc about this error argument?  It looks like the intent is to allow some error handling to be delegated down to the `VisitorFunc`, but if that's the case, it's a little unclear to me why I wouldn't have something more like

```
type Visitor interface{
    // In the normal path, Visit is called with an info
    Visit(*Info) error

    // If an error occurs before Visit is called, HandleError is invoked.  A nil return indicates a desire to continue to the next time to be visited.
    // This is useful for ignoring certain kinds of errors or aggregating errors in some way.
    HandleError(error) error
```